### DISCUSSION TITLE
Update store state through a ref

### DISCUSSION THREAD

**Discussion by divmgl** [pmndrs/zustand#2081] [Q&A] [answered]
Hi. I'm passing in a `ref` to my Zustand store so that whenever the parent component receives new information from its props it calls the `ref` and then updates the store. Doesn't seem to be working though. Any ideas why?

Here's a small snippet of code that should illustrate the issue. 

```tsx
// createStoreContext is a port of createContext.
const { Provider, useStore: useTaskListStore } = createStoreContext<TaskListStore>()

// The provider
function TaskListProvider({ children, ...props }: PropsWithChildren<TaskListV2Props>) {
  const taskRef = useRef<ProjectTasksUpdater | null>(null)

  useEffect(() => {
    taskRef.current?.(props.tasks)
  }, [props.tasks])

  return <Provider createStore={() => createTaskListStore(props, taskRef)}>{children}</Provider>
}

// Inside of the store
function createTaskListStore(
  props: TaskListV2Props,
  taskRef: MutableRefObject<ProjectTasksUpdater | null>
) {
  return createStore<TaskListStore>()((set) => {
    taskRef.current = (newTasks: ProjectTask[]) => {
      set({ tasks: newTasks })
    }
```

What's interesting is that I'm seeing the state is updating internally (when I call `get`) but the children are not being rerendered. In the children, I'm grabbing the state using:

```tsx
const tasks = useTaskListStore((state) => state.tasks)
```

Thanks.

**Comment by divmgl** [comment, marked as answer]
The bug was in `createStoreContext`, which is unrelated to Zustand.

**Reply by dbritto-dev** [reply]
@divmgl the implementation and usage of `useRef` in this case is wrong.

> *Note:* `useRef` does not trigger any render so even if `get` is called you won't see any updates

**Reply by divmgl** [reply]
Hm, I'm unsure that's the case now. Doesn't `set` in Zustand trigger an update? It's currently working fine on our end (after the very big problem with `createStoreContext`, that is not detailed here).
